src/particles/elements/Drift.H

```diff
@@ -53,17 +53,27 @@ namespace impactx
             amrex::ParticleReal const y = p.pos(1);
             amrex::ParticleReal const t = p.pos(2);
 
+            // intialize output values of momenta
+            amrex::ParticleReal pxout = px;
+            amrex::ParticleReal pyout = py;
+            amrex::ParticleReal ptout = pt;
+
             // access reference particle values to find beta*gamma^2
             amrex::ParticleReal const pt_ref = refpart.pt;
             amrex::ParticleReal const betgam2 = pow(pt_ref, 2) - 1.0_prt;
 
             // advance position and momentum (drift)
             p.pos(0) = x + m_ds * px;
-            // px = px;
+            // pxout = px;
             p.pos(1) = y + m_ds * py;
-            // py = py;
+            // pyout = py;
             p.pos(2) = t + (m_ds/betgam2) * pt;
-            // pt = pt;
+            // ptout = pt;
+
+            // assign updated momenta
+            px = pxout;
+            py = pyout;
+            pt = ptout;
```

Comment:
I cannot see where we change the values for `px` or `pxout` in this function (or `y` or `t`), is this really needed for the drift?

Reply:
No, this is not needed for the drift.  I made this change so that the logic of all of the elements is the same.  If someone uses an existing element as the template for a new one, then this could avoid confusion.

Reply:
I see. We have to check if the compiler identifies and removes those extra copies.

Otherwise, we might want to remove the unneeded ones and/or create a custom element that we want to use as template?

Reply:
Discussed today:
- current structure is nice because if we comment the code for momentum in between, we keep the identity map (for playing/debugging/new elements)
- the compiler recognized the overwritten initial assignment and discards it in any optimization mode >= `-O1`: https://godbolt.org/z/1Ecnr53oE